doc/getting-started.md

```diff
@@ -21,59 +19,72 @@ deployment "istio-mixer" created
 
 ```
+# The CouchDB deployment must join the Istio service mesh.  Note that the pod will be recreated.
+kubectl get deployment couchdb --output yaml | istioctl kube-inject --filename - --tag 2017-03-22-17.30.06 | kubectl apply -f -
+```
 
-# Backend "hello" service with an instance of container gcr.io/istio-testing/runtime:demo
-kubectl create -f <(istioctl kube-inject -f doc/hello-and-proxy.yaml)
+For this tutorial an NGINX front-end will talk to CouchDB.  To enable communication we must expose the deployment as a Kubernetes Service.
 
-# "Frontend" service with an instance of container gcr.io/istio-testing/runtime:demo
-kubectl create -f <(istioctl kube-inject -f doc/frontend-and-proxy.yaml)
-kubectl expose -f <(istioctl kube-inject -f doc/frontend-and-proxy.yaml) --type=NodePort --name=frontend
+```
+kubectl expose deployment couchdb --name=couchdb
+# Let Istio know that the port is HTTP, not generic TCP
+kubectl patch service couchdb --type='json' -p='[{"op": "add", "path": "/spec/ports/0/name", "value":"http"}]'
+```
+
+Because the CouchDB service we created is not exposed to the outside world for security reasons
+we cannot test it from outside the cluster.  To verify that it is working correctly, we can run
+a _curl_ command against couchdb:5984 *from inside the cluster* using the public _dockerqa/curl_
+image from the Docker hub:
 
 ```
+kubectl run -i --rm --restart=Never dummy --image=dockerqa/curl:ubuntu-trusty --command -- curl couchdb:5984
+```
 
-<!---
-Note that to stand up the frontend without Istio, we can do
-kubectl run frontend --image=gcr.io/google-samples/hello-frontend:1.0 --port=8080 --overrides='{ "apiVersion": "extensions/v1beta1", "spec": { "containers": { "lifecycle": { "preStop": { "exec": { "command": ["/usr/sbin/nginx","-s","quit"] } } } } } }'
-# or
-kubectl run front-no-istio --image=gcr.io/google-samples/hello-frontend:1.0 --port=80
-# and then
-kubectl expose deployment front-no-istio --type=NodePort --name=frontend-no-istio
-kubectl describe services frontend-no-istio
--->
+The output should include '{"couchdb":"Welcome"...'
 
-At this point we have two microservices.  Let us test by looking up the public port
+Next we will start NGINX and configure it to proxy for the CouchDB service started in the previous step
 
 ```
-# Get the IP address.  (Alternate: gcloud compute instances list)
-minikube ip
-export IP=`minikube ip`
-# Get the NodePorts for both original and Istio versions
-kubectl describe services frontend
-export NODEPORT=...
+# Create an NGINX service
+kubectl run nginx --image=docker.io/nginx
+# The NGINX deployment must join the Istio service mesh.
+kubectl get deployment nginx --output yaml | istioctl kube-inject --filename - --tag 2017-03-22-17.30.06 | kubectl apply -f -
+# Configure the NGINX to point to the CouchDB.  Note that this step is just for the demo.  You should
+# not make changes to running containers in production.  This change is lost if the pod restarts.
```

Comment:
This is a getting started document. In that spirit, having a demo style setup (hacking the nginx config online) seems weird. Either way, if there is a python/nodejs app in front of couch db, it would solve the issue. It would also allow you to get rid of the "is couchdb working" paragraph above. Instead, you could add a "is my app working after adding Istio".